crates/autopilot/src/solvable_orders.rs

```diff
@@ -269,8 +273,23 @@ impl SolvableOrdersCache {
 }
 
 /// Filters all orders whose owners are in the set of "banned" users.
-fn filter_banned_user_orders(mut orders: Vec<Order>, banned_users: &HashSet<H160>) -> Vec<Order> {
-    orders.retain(|order| !banned_users.contains(&order.metadata.owner));
+async fn filter_banned_user_orders(
+    mut orders: Vec<Order>,
+    banned_users: &banned::Users,
+) -> Vec<Order> {
+    let futures = orders.drain(..).map(|order| async move {
+        (
+            order.clone(),
+            banned_users.is_banned(order.metadata.owner).await,
+        )
+    });
+
+    for (order, banned) in join_all(futures).await {
+        // If the user is not banned or the check failed, keep the order.
+        if banned.is_err() || banned.is_ok_and(|banned| !banned) {
+            orders.push(order)
+        }
+    }
     orders
 }
```

Comment:
nit: cloning could be avoided here with something like
```suggestion
async fn filter_banned_user_orders(orders: Vec<Order>, banned_users: &banned::Users) -> Vec<Order> {
    let futures = orders.into_iter().map(|order| async move {
        banned_users
            .is_banned(order.metadata.owner)
            .await
            .map_or(true, |banned| !banned)
            .then_some(order)
    });
    join_all(futures).await.into_iter().flatten().collect()
}
```